homeassistant/components/mystrom/sensor.py

```diff
@@ -23,33 +24,43 @@
 
 
 @dataclass(frozen=True)
-class MyStromSwitchSensorEntityDescription(SensorEntityDescription):
-    """Class describing mystrom switch sensor entities."""
+class MyStromSensorEntityDescription(SensorEntityDescription):
+    """Class describing mystrom switch and PIR sensor entities."""
 
-    value_fn: Callable[[MyStromSwitch], float | None] = lambda _: None
+    value_fn: Callable[[MyStromPir | MyStromSwitch], float | None] = lambda _: None
 
 
-SENSOR_TYPES: tuple[MyStromSwitchSensorEntityDescription, ...] = (
-    MyStromSwitchSensorEntityDescription(
+SENSOR_TYPES: tuple[MyStromSensorEntityDescription, ...] = (
+    MyStromSensorEntityDescription(
```

Comment:
Okay so instead of moving everything to a getattr (which we generally don't like as its not type safe), why don't we move to a setup where we add a list of objects (`MyStromPir | MyStromSwitch`) to every description which says when it should be created. I also think we can split the temperature one

Reply:
I agree that the types should be preserved, but couldn't find any integration with this particular pattern. So I just tried to implemented it with a long lambda function.